### PR TITLE
Add missing postbuild events; fix query class name in Engine

### LINKED ISSUE #10: Missing postbuild events
#### Description:

Missing build events in the template projects.

### LINKED ISSUE #11: Incorrect Query partial class name (Create)

### PR DESCRIPTION
**Pull request by alelom**
Closes #10 
Closes #11

### CODE REVIEW

File: SoftwareName_Engine/Query/ExampleQueryMethod.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
 namespace BH.Engine.Adapters.SoftwareName
 {
-    public static partial class Create
+    public static partial class Query
     {
         /***************************************************/
         /**** Public Methods                            ****/
```
Comment (IsakNaslundBh) [inline SoftwareName_Engine/Query/ExampleQueryMethod.cs:36]: Cant comment on the line below, but might as well fix the method name, and making it extension while you are on this (as you noted yesterday 👍 )

basically, change from:

`public static string ExampleCreateMethod(ExampleObject exampleObject)`

to

`public static string ExampleQueryMethod(this ExampleObject exampleObject)`

**Review by IsakNaslundBh — CHANGES_REQUESTED** [review]
Initial issues resolved, but think we might as well add the last small fix to the Query name and make extension :)

File: SoftwareName_Engine/Query/ExampleQueryMethod.cs (hunk 1 above)
Comment (alelom) [reply SoftwareName_Engine/Query/ExampleQueryMethod.cs:36]: Oh you're right thanks!

File: SoftwareName_Engine/Query/ExampleQueryMethod.cs [hunk 2]
```diff
@@ -30,20 +30,25 @@
...
         /***************************************************/
 
         [Description("Description of the method. Will appear in the UI tooltip.")]
         [Input("exampleObject", "Description of the input. Will appear in the UI tooltip.")]
+        [Input("additionalInput", "Description of the input. Will appear in the UI tooltip.")]
         [Output("outputName", "Description of the output. Will appear in the UI tooltip.")]
-        public static string ExampleCreateMethod(ExampleObject exampleObject)
+        public static string ExampleCreateMethod(this ExampleObject exampleObject, string additionalInput = "")
```
Comment (IsakNaslundBh) [inline SoftwareName_Engine/Query/ExampleQueryMethod.cs]:
```suggestion
        public static string ExampleQueryMethod(this ExampleObject exampleObject, string additionalInput = "")
```

**Review by IsakNaslundBh — CHANGES_REQUESTED** [review]
Method name as well :)

File: SoftwareName_Engine/Query/ExampleQueryMethod.cs (hunk 2 above)
Comment (alelom) [reply SoftwareName_Engine/Query/ExampleQueryMethod.cs]: Woops, sorry! Keep on missing things

**Comment by IsakNaslundBh** [comment]
@BHoMBot check code compliance
@BHoMBot check document compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

**Review by FraserGreenroyd — APPROVED** [review]
Following discussion with @IsakNaslundBh offline, I am happy to approve this PR with the failing project-compliance check because all other checks have passed, indicating the starting template is compliant. The project-compliance will be checked when it gets uploaded in new repositories and corrected at that stage as necessary so does not make sense to make correct in this template.
 
image

**Review by IsakNaslundBh — APPROVED** [review]
Things looking good now!

**Pull request merged**